tests/suites/test_suite_sm3.function

```diff
@@ -0,0 +1,75 @@
+/* BEGIN_HEADER */
+#include "mbedtls/sm3.h"
+/* END_HEADER */
+
+/* BEGIN_CASE depends_on:MBEDTLS_SM3_C */
+void sm3_valid_param( )
+{
+    TEST_VALID_PARAM( mbedtls_sm3_free( NULL ) );
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_SM3_C:MBEDTLS_CHECK_PARAMS:!MBEDTLS_PARAM_FAILED_ALT */
+void sm3_invalid_param( )
+{
+    mbedtls_sm3_context ctx;
+    unsigned char buf[64] = { 0 };
+    size_t const buflen = sizeof( buf );
+
+    TEST_INVALID_PARAM( mbedtls_sm3_init( NULL ) );
+
+    TEST_INVALID_PARAM( mbedtls_sm3_clone( NULL, &ctx ) );
+    TEST_INVALID_PARAM( mbedtls_sm3_clone( &ctx, NULL ) );
+
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_SM3_BAD_INPUT_DATA,
+                            mbedtls_sm3_starts_ret( NULL ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_SM3_BAD_INPUT_DATA,
+                            mbedtls_sm3_starts_ret( &ctx ) );
+
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_SM3_BAD_INPUT_DATA,
+                            mbedtls_sm3_update_ret( NULL, buf, buflen ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_SM3_BAD_INPUT_DATA,
+                            mbedtls_sm3_update_ret( &ctx, NULL, buflen ) );
+
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_SM3_BAD_INPUT_DATA,
+                            mbedtls_sm3_finish_ret( NULL, buf ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_SM3_BAD_INPUT_DATA,
+                            mbedtls_sm3_finish_ret( &ctx, NULL ) );
+
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_SM3_BAD_INPUT_DATA,
+                            mbedtls_internal_sm3_process( NULL, buf ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_SM3_BAD_INPUT_DATA,
+                            mbedtls_internal_sm3_process( &ctx, NULL ) );
+
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_SM3_BAD_INPUT_DATA,
+                            mbedtls_sm3_ret( NULL, buflen, buf ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_SM3_BAD_INPUT_DATA,
+                            mbedtls_sm3_ret( buf, buflen, NULL ) );
+    TEST_INVALID_PARAM_RET( MBEDTLS_ERR_SM3_BAD_INPUT_DATA,
+                            mbedtls_sm3_ret( buf, buflen, buf ) );
```

Comment:
Base on #4516, we can remove above NULL pointer check

Reply:
Good point
Note: I ended up removing all the TEST_INVALID_PARAM_RET(), so the whole sm3_invalid_param() had to go because the generate script generated an unused "exit:" label.